### PR TITLE
Initial fix for logging multuthreaded corruption

### PR DESCRIPTION
**Pull request by leekillough**
Fixes SWDEV-215335

Right now it does not work, because the compiler/linker/runtime is inexplicably not calling `rocblas_create_handle()` from `rocblas_local_handle::rocblas_local_handle()`.

### CODE REVIEW

File: library/src/rocblas_auxiliary.cpp [hunk 1]
```diff
@@ -60,56 +58,52 @@ extern "C" rocblas_status rocblas_set_pointer_mode(rocblas_handle handle, rocbla
...
-    }
-    catch(...)
-    {
-        return rocblas_status_internal_error;
-    }
     return rocblas_status_success;
 }
+catch(...)
+{
+    return rocblas_status_memory_error;
```
Comment (YvanMokwinski) [inline library/src/rocblas_auxiliary.cpp]: Are you sure that that it should be a rocblas_status_memory_error rather than rocblas_status_internal_error?

Comment (leekillough) [reply library/src/rocblas_auxiliary.cpp]: It almost always catches a `std::bad_alloc` from the `new` when it receives an exception, so the memory error is more descriptive than internal error. I could catch        `std::bad_alloc` and flag it as memory error, while making other exceptions return internal error.

Comment (TorreZuk) [reply library/src/rocblas_auxiliary.cpp]: Normally I think that would be good enough but for this case I think  two catch blocks might be better to discern the two failure modes.  Or two try catches.   I think we may want another error enum but not for this case, discuss later today.

Comment (leekillough) [reply library/src/rocblas_auxiliary.cpp]: See #863 for a general solution to the exception issue. If you need to catch different exceptions and make them return a different `rocblas_status`, then change `library/src/include/utility.h`, which has a universal exception -> `rocblas_status` conversion function.

**Comment by leekillough** [comment]
Will have to reopen in 2020

**Pull request closed without merging**